finality-aleph/src/sync/substrate/verification/mod.rs

```diff
@@ -0,0 +1,114 @@
+use std::{
+    fmt::{Display, Error as FmtError, Formatter},
+    marker::PhantomData,
+    sync::Arc,
+};
+
+use aleph_primitives::BlockNumber;
+use codec::Encode;
+use sc_client_api::HeaderBackend;
+use sp_runtime::traits::{Block as BlockT, Header as SubstrateHeader};
+
+use crate::{
+    session_map::AuthorityProvider,
+    sync::{
+        substrate::{
+            verification::{
+                cache::{CacheError, VerifierCache},
+                verifier::SessionVerificationError,
+            },
+            Justification,
+        },
+        Verifier,
+    },
+};
+
+mod cache;
+mod verifier;
+
+pub use verifier::SessionVerifier;
+
+/// Supplies finalized number. Will be unified together with other traits we used in A0-1839.
+pub trait FinalizationStatus {
```

Comment:
if the naming will be different after unification ignore. I don't get the name of this trait with Status. Its rather smth like FInalizationInfo or smt like this.

Reply:
I intended it to be more aligned with `ChainStatus` struct. `FinalizationInfo` is also fine by me.